packages/web/src/common/store/buy-audio/slice.ts

```diff
@@ -0,0 +1,186 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { JupiterTokenSymbol } from 'common/store/buy-audio/constants'
+
+export enum Flow {
+  COINBASE_PAY = 'COINBASE_PAY'
+}
+
+export enum QuoteStatus {
+  IDLE = 'IDLE',
+  QUOTING = 'QUOTING',
+  SUCCEEDED = 'SUCCEEDED',
+  FAILED = 'FAILED'
+}
+
+export enum ExchangeStatus {
+  IDLE = 'IDLE',
+  WAITING = 'WAITING',
+  EXCHANGING = 'EXCHANGING',
+  SUCCEEDED = 'SUCCEEDED',
+  FAILED = 'FAILED'
+}
+
+type JupiterSwapActionPayload = {
+  inputTokenSymbol: JupiterTokenSymbol
+  outputTokenSymbol: JupiterTokenSymbol
+  inputAmount: number
+  forceFetch?: boolean
+  slippage?: number
+  padForSlippage?: boolean
+}
+
+type BuyAudioState = {
+  flow: Flow
+  swaps: Record<
+    JupiterTokenSymbol,
+    Record<
+      JupiterTokenSymbol,
+      {
+        inputUiAmount?: string
+        outputUiAmount?: string
+        quoteStatus: QuoteStatus
+        exchangeStatus: ExchangeStatus
+      }
+    >
+  >
+}
+
+const initialState: BuyAudioState = {
+  flow: Flow.COINBASE_PAY,
+  swaps: {}
+}
+
+const initSwapStateIfNecessary = (
+  state: BuyAudioState,
+  inputTokenSymbol: JupiterTokenSymbol,
+  outputTokenSymbol: JupiterTokenSymbol
+) => {
+  if (
+    !state.swaps[inputTokenSymbol] ||
+    !state.swaps[inputTokenSymbol][outputTokenSymbol]
+  ) {
+    state.swaps[inputTokenSymbol] = {
+      ...state.swaps[inputTokenSymbol],
+      [outputTokenSymbol]: {
+        quoteStatus: QuoteStatus.IDLE,
+        exchangeStatus: ExchangeStatus.IDLE
+      }
+    }
+  }
+  return state
+}
+
+const slice = createSlice({
+  name: 'ui/buy-audio',
+  initialState,
+  reducers: {
+    exchangeAfterBalanceChange: (
+      state,
+      {
+        payload: { inputTokenSymbol, outputTokenSymbol }
+      }: PayloadAction<JupiterSwapActionPayload>
+    ) => {
+      state = initSwapStateIfNecessary(
+        state,
+        inputTokenSymbol,
+        outputTokenSymbol
+      )
+      state.swaps[inputTokenSymbol][outputTokenSymbol].exchangeStatus =
+        ExchangeStatus.WAITING
+    },
+    exchange: (
```

Comment:
i think this is fine, but would it be helpful to know what we are exchanging? seems like... exchanging tokens? i guess exchanging is fine, but yeah just thinkin